DataDog/datadog-agent
# [NPM-3080] Fix TestBlockingReadCounts

### What does this PR do?

Another attempt to fix `TestBlockingReadCounts` by:
* make socket blocking
* use `syscall.RawConn` to ensure socket fd is valid for socket operations

### Motivation



### Additional Notes



### Possible Drawbacks / Trade-offs



### Describe how to test/QA your changes



### Reviewer's Checklist
